cdap-docs/reference-manual/source/http-restful-api/metrics.rst

```diff
@@ -371,161 +417,205 @@ To query a metric within a given context, perform an HTTP POST request::
-  namespace of a CDAP instance::
+  {"startTime":<start-time>, "endTime":<end-time>, "series":<series-array>}
 
-    POST '<base-url>/metrics/query?context=namespace.default.app.CountRandom.flow.*
-      &metric=system.process.events.processed&start=now-5s&count=5'
-
-    POST '<base-url>/metrics/query?context=namespace.default.app.CountRandom
-      &metric=system.process.events.processed&start=now-5s&count=5'
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
 
-    POST '<base-url>/metrics/query?context=namespace.default
-      &metric=system.process.events.processed&start=now-5s&count=5'
+   * - Name
+     - Description
+   * - ``<start-time>``
+     - Start time, in seconds, with 0 being from the beginning of the query records
+   * - ``<metric>``
+     - End time, in seconds
+   * - ``<series-array>``
+     - An array of metric results, which can be one series, a multiple time series, or
+       none (an empty array)
 
-- To request user-defined metrics instead of system metrics, specify ``user`` instead of ``cdap`` in the URL
-  and specify the user-defined metric at the end of the request.
+If a particular metric has no value, a query will return an empty array in the
+``"series"`` of the results, such as::
 
-  For example, to request a user-defined metric for the *HelloWorld* Application's *WhoFlow* Flow::
+  {"startTime":0,"endTime":1429475995,"series":[]}
+    
+You can also receive such a result from querying a metric that does not exist, either
+because it does not exist at the context given or if the query is incorrectly
+formulated:
+  
+  ``...metric=user.names.bytes?aggregate=true``
+  
+will return the empty result, as the metric name will be interpreted as
+``"user.names.bytes?aggregate=true"`` instead of ``"user.names.bytes"``.
 
-    POST '<base-url>/metrics/query?context=namespace.default.app.HelloWorld.flow.WhoFlow.flowlet.saver
-      &metric=user.names.bytes&aggregate=true'
 
 .. _http-restful-api-v3-metrics-multiple:
 .. _http-restful-api-metrics-multiple:
 
 Querying for Multiple Metrics
------------------------------
-Retrieving multiple metrics at once, by issuing an HTTP POST request with a JSON list as
-the request body that enumerates the name and attributes for each metric, is currently not
-supported in this API. Instead, use the v2 API until it is supported in a future release.
+.............................
+
+Retrieving multiple metrics at once can be accomplished by issuing an HTTP POST request
+with a JSON list as the request body that enumerates the name and attributes for each
+metric. The format of the request and the JSON body depends on whether the metrics share
+the same context or are being called for different contexts. 
+
+In both cases, queries are identified by a ``<query-id>`` (in the examples below,
+*query1*, *query2*, *eventsIn*, *eventsOut*); the ``<query-id>`` is then used in the
+returned result to identify the series.
```

Comment:
description is incorrect, we don't have query-id's when when we query multiple metrics in the same context using the url and not the body. only when the queries are batched and specified in the body of request, we use query-ids.